x-pack/plugins/infra/public/components/asset_details/links/link_to_alerts.tsx

```diff
@@ -9,7 +9,7 @@ import { FormattedMessage } from '@kbn/i18n-react';
 import { EuiButtonEmpty } from '@elastic/eui';
 
 export interface LinkToAlertsRuleProps {
-  onClick?: () => void;
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
I think it'd be better if you handled the `e.stopPropagation();` directly in this component instead of passing up the element. We guarantee the same behaviour for all clients then. WDYT?

```js
onClick={(e) => {
   e.stopPropagation();
   onClick();
}}
```

Reply:
Here we need this only because inside the accordion header (in the previous case or any other cases this is not needed) - [this example](https://eui.elastic.co/#/layout/accordion#interactive-content-in-the-trigger). Also `onClick` is `optional` parameter so [here](https://github.com/elastic/kibana/blob/da34e181cd8bdde7bc4a58c08f5bc9b4ef715360/x-pack/plugins/infra/public/components/asset_details/hooks/use_page_header.tsx#L98) in the header, for example, we don't need it.